speeduino/decoders.ino

```diff
@@ -648,50 +704,95 @@ void triggerThird_missingTooth(void)
   } // Trigger filter
 }
 
-uint16_t getRPM_missingTooth(void)
-{
+uint16_t getRPM_missingTooth(void){
   uint16_t tempRPM = 0;
-  if( currentStatus.RPM < currentStatus.crankRPM )
+  uint32_t timeInterval;
+  uint8_t x;
+  const uint8_t amountOfEdges=16; //amount of edges to use. Good to use power ot 2 values(4,8,16 or 32), this gives better performance (compiler optimizes division to bit shift)
+  uint8_t tempToothCurrentCount = (uint8_t)toothCurrentCount; //8bits , then there is no race condition for read.
+  //calculate RPM over 16 last seen tooth (takes account the cap of the missing tooth)
+  // caclulate circular buffer index x, this gives us the position in the buffer where needed element is to be found.
+  // Take advantage of the fact that missing tooth(teeth) are always just before the buffer index(ToothCurrentCount) rollback.
+  if(tempToothCurrentCount > amountOfEdges)
```

Comment:
It is not entirely that bad, because it is a sliding window. So it takes basically average of 5 seconds back from the currently last seen tooth. The notice is valid though. It may be too slow on the low res trigger. 
The solution I see here is to not have fixed value for the used amountOfEdges, but to make the value smaller for smaller tooth count wheels.
Thanks

Reply:
(average of last 5 seconds at 60RPM with 4-1 arrangement that is)